app/legacy_lib/mailchimp.rb

```diff
@@ -49,6 +49,11 @@ def self.signup supporter, mailchimp_list_id
 		put("/lists/#{mailchimp_list_id}/members", @options.merge(:body => body_hash.to_json))
   end
 
+  def self.signup_nonprofit_user(drip_email_list, nonprofit, user)
+    body_hash = @body.merge(create_nonprofit_user_subscribe_body(nonprofit, user))
+    put(drip_email_list.list_members_path, @options.merge(:body => body_hash.to_json))
+  end 
```

Comment:
You'll need a spec to verify that it works. You will need to mock the `put` call. I can pair with you on figuring out the best way to do that.